test/runner/tob.ts

```diff
@@ -0,0 +1,144 @@
+import { printSourceFile } from '../../src/compiler/printer';
+import { parseScript, parseModule } from '../../src/kataw';
+import { promiseToReadFile, promiseToWriteFile, Constants, deepEqual } from './utils';
+
+// testing object
+export interface Tob {
+  filename: string;
+  content?: string;
+  input: string;
+  parserOptions: any;
+  printerOptions: any;
+  cst?: any;
+  $cst?: any;
+  printed?: any;
+  $printed?: any;
+  diagnostics?: any;
+  $diagnostics?: any;
+  isMatched?: boolean;
+}
+
+export function updateTob(tob: any, updateItem: any): void {
+  (updateItem === 'all' || updateItem === 'parser') && (tob.cst = tob.$cst);
```

Comment:
Update means "overwrite". So the 'tob.cst' has to contain he new generated ast, and not the old ast that should be overwritten.

Reply:
so you mean: do not overwrite the parsed cst from the md files?

Reply:
need to overwrite it totally, but it also add info from printer, and also from previous cst and mix it with new output before writing it all together to a file.